atmat/attrack/elempass.m

```diff
@@ -19,13 +19,22 @@
 %
 % See also: RINGPASS, LINEPASS
 
-[props.Energy,varargs]=getoption(varargin,'Energy',0.0);
+props = struct;
+[varenergy,varargs]=getoption(varargin,'Energy',0.0);
 [particle,varargs]=getoption(varargs,'Particle',[]);
 [methodname,varargs]=getoption(varargs,'PassMethod',elem.PassMethod); %#ok<ASGLU>
 if ~isempty(particle)
     props.Particle=particle;
 end
 
+% check pass methods that require energy
+needs_energy = {'GWigSymplecticPass', 'GWigSymplecticRadPass'};
+if any(strcmp(needs_energy,methodname)) && varenergy == 0
+    error('Energy needs to be defined');
```

Comment:
I think that all the *RadPass methods need energy. Plus the wiggler,  because of the absolute field definition in T… And also RFCavityPass.

Unfortunately, the pass method names must be hard-coded. So It has to be revised when new pass methods are introduced.

But I guess it's better than nothing. So OK for me.